src/cellularnoise.rs

```diff
@@ -15,76 +16,75 @@ fn noise_gen(
     height_as_str: &str,
 ) -> Result<String> {
     let percentage = percentage_as_str.parse::<i32>()?;
-    let smoothing_level = smoothing_level_as_str.parse::<i32>()?;
-    let birth_limit = birth_limit_as_str.parse::<i32>()?;
-    let death_limit = death_limit_as_str.parse::<i32>()?;
+    let smoothing_level = smoothing_level_as_str.parse::<usize>()?;
+    let birth_limit = birth_limit_as_str.parse::<usize>()?;
+    let death_limit = death_limit_as_str.parse::<usize>()?;
     let width = width_as_str.parse::<usize>()?;
     let height = height_as_str.parse::<usize>()?;
-
-    // Noise generation
-    let mut zplane = vec![vec![false; width]; height];
-    for row in zplane.iter_mut() {
-        for cell in row.iter_mut() {
-            *cell = rand::thread_rng().gen_range(0..100) < percentage;
-        }
-    }
-
-    // Smoothing part
-    for _timer in 0..smoothing_level {
-        let zplane_old = zplane.clone();
-        for i in 0..height {
-            for j in 0..width {
-                let mut sum = 0;
-
-                if i > 0 {
-                    if j > 0 {
-                        sum += if zplane_old[i - 1][j - 1] { 1 } else { 0 };
+    //we populate it
+    let mut filled_vec = (0..height + 3)
```

Comment:
comment explaining the +3s? i get it, but it's not immediately obvious

Reply:
good idea